### PR TITLE
Refactor Tenants Tests

### PR DESCRIPTION
**Pull request by ericholguin**
This PR refactors **v4/tenants_test.go**, **v3/tenants_test.go**, in order to make tests more granular, providing transparency on what is being tested and reducing the need to dig through test code to confirm what was tested. In addition this new test structure should also make it easier for new tests to be added as well as reduce redundant code.

- Adds data driven testing
    - Descriptive test names
- Sub tests for organization
    - Use GoLangs test runner
- Use assert functionality
    - Provides fundamental assertions
    - Streamlines code
    - Reduces nested conditionals
- Adds expectation functions
     - Test specific expectations in a clear concise way
     - Reusable expectations

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops Tests

## What is the best way to verify this PR?

Run TO Integration tests

## PR submission checklist
- [x] This PR has tests 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/testing/api/v4/tenants_test.go [hunk 1]
```diff
@@ -16,546 +16,323 @@ package v4
...
-	response, _, err := TOSession.UpdateTenant(tn.ID, tn, client.RequestOptions{})
-	if err != nil {
-		t.Fatalf("cannot update Tenant: %v - alerts: %+v", err, response.Alerts)
+func GetTenantID(t *testing.T, name string) func() int {
+	return func() int {
+		opts := client.NewRequestOptions()
+		opts.QueryParameters.Set("name", name)
+		tenants, _, err := TOSession.GetTenants(opts)
+		assert.RequireNoError(t, err, "Get Tenants Request failed with error:", err)
+		assert.RequireEqual(t, 1, len(tenants.Response), "Expected response object length 1, but got %d", len(tenants.Response))
```
Comment (srijeet0406) [inline traffic_ops/testing/api/v4/tenants_test.go:308]: If `len(tenants.Response)` is not 1 for some reason, does this stop the test here? Or does it still proceed to return `tenants.Response[0].ID`? If it is the former, then we're fine. If not, then the `return` statement could cause a panic.

File: traffic_ops/testing/api/v4/tenants_test.go [hunk 2]
```diff
@@ -16,546 +16,323 @@ package v4
...
-	modTenant.ParentID = modTenant.ID
-	_, reqInf, err := TOSession.UpdateTenant(modTenant.ID, modTenant, client.RequestOptions{})
-	if err == nil {
-		t.Fatalf("expected an error when trying to update the 'root' tenant, but got nothing")
-	}
-	if reqInf.StatusCode != http.StatusBadRequest {
-		t.Errorf("expected a status 400 Bad Request, but got %d", reqInf.StatusCode)
+func validateTenantSort() utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, resp interface{}, alerts tc.Alerts, _ error) {
+		assert.RequireNotNil(t, resp, "Expected Tenant response to not be nil.")
```
Comment (srijeet0406) [inline traffic_ops/testing/api/v4/tenants_test.go:246]: If `resp` is `nil`, does it stop the test there? If yes, we should be good.

File: traffic_ops/testing/api/v3/tenants_test.go [hunk 3]
```diff
@@ -16,352 +16,232 @@ package v3
...
+			tenants = append(tenants, tenant.Name)
 		}
+		assert.Equal(t, true, sort.StringsAreSorted(tenants), "List is not sorted by their names: %v", tenants)
 	}
 }
 
-func ExtractXMLID(ds *tc.DeliveryServiceNullable) string {
-	if ds.XMLID != nil {
-		return *ds.XMLID
+func GetTenantID(t *testing.T, name string) func() int {
```
[the remark this answers is not in the document]
Comment (ericholguin) [reply traffic_ops/testing/api/v3/tenants_test.go:216]: Needs to be a function for the endpoint Id field, before it was only being used in a request body.

File: traffic_ops/testing/api/v4/tenants_test.go [hunk 4]
```diff
@@ -16,546 +16,323 @@ package v4
...
-		t.Fatalf("Expected exactly one Tenant to exist with the name '%s', found: %d", name, len(resp.Response))
-	}
-	tn := resp.Response[0]
-
-	tn.Active = active
-	response, _, err := TOSession.UpdateTenant(tn.ID, tn, client.RequestOptions{})
-	if err != nil {
-		t.Fatalf("cannot update Tenant: %v - alerts: %+v", err, response.Alerts)
+func GetTenantID(t *testing.T, name string) func() int {
+	return func() int {
```
[the remark this answers is not in the document]
Comment (ericholguin) [reply traffic_ops/testing/api/v4/tenants_test.go:303]: Same as above

File: traffic_ops/testing/api/v4/tenants_test.go (hunk 1 above)
Comment (ericholguin) [reply traffic_ops/testing/api/v4/tenants_test.go:308]: Yes it'll stop the test there. Require* functions call t.Fatal which stops test execution.

**Review by srijeet0406 — APPROVED** [review]

**Pull request merged**